src/Memory.cs

```diff
@@ -131,7 +136,7 @@ public string ReadNullTerminatedString(IStore store, int address)
                 throw new InvalidOperationException("string is not null terminated");
             }
 
-            return Encoding.UTF8.GetString(slice.Slice(0, terminator));
+            return MemoryEncoding.GetString(slice.Slice(0, terminator));
```

Comment:
I think we should keep `ReadNullTerminatedString` specific to UTF-8 given that we can't can't simply `IndexOf` for the null byte in other encodings.